### PR TITLE
Add destructors for all of the layers

### PR DESCRIPTION
**Pull request by DLu**
Add destructors for all the layers so that the dynamic parameter clients are destroyed as well.

### CODE REVIEW

**Comment by mikeferguson** [comment]
@DLu is your intent to merge this before or after a new release?

File: costmap_2d/include/costmap_2d/footprint_layer.h [hunk 1]
```diff
@@ -53,6 +53,7 @@ class FootprintLayer : public Layer
 {
 public:
   virtual void onInitialize();
+  ~FootprintLayer();
```
Comment (mikeferguson) [inline costmap_2d/include/costmap_2d/footprint_layer.h]: Should be virtual, no?

Comment (DLu) [inline costmap_2d/include/costmap_2d/footprint_layer.h]: Probably. Should they all be?

**Comment by DLu** [comment]
@mikeferguson I was intending for after the release, since this is mostly a precursor to my next pull request that runs the clear_costmap tests, but we can push it through before if you want.

File: costmap_2d/include/costmap_2d/footprint_layer.h (hunk 1 above)
Comment (DLu) [inline costmap_2d/include/costmap_2d/footprint_layer.h]: Fixed all.

**Comment by mikeferguson** [comment]
Ok, I'm going to do a release right now -- and then we can merge this later today.

**Pull request merged**